### PR TITLE
Separated CQL ser/deser code from driver logic

### PR DESCRIPTION
**Pull request by wprzytula**
Another crate was created in order to allow reuse of low-level code
in other projects (including under-progress proxy). The whole `frame`
module was made separate, along with structs regarding transport
errors. Some tests that require driver transport logic as dependency
were moved from `frame` to `transport`.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

**Comment by psarna** [comment]
Just to clarify - this PR is only about code movement, no logic was changed, right?

**Comment by wprzytula** [comment]
> Just to clarify - this PR is only about code movement, no logic was changed, right?

I confirm. The sole purpose is to avoid circular dependency after adding a proxy crate, because the driver will depend on it for tests (and proxy would have to depend on driver to use its cql ser/deser code).

**Review by psarna — APPROVED** [review]
@piodul any objections? Looks good to me

**Review by piodul — COMMENTED** [review]
Please try to slim down the dependencies in the new `scylla-cql` crate. Apart from that, LGTM.

File: scylla-cql/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+description = "CQL data types and primitives, for interacting with Scylla."
+repository = "https://github.com/scylladb/scylla-rust-driver"
+readme = "../README.md"
+keywords = ["database", "scylla", "cql", "cassandra"]
+categories = ["database"]
+license = "MIT OR Apache-2.0"
+
+[features]
+defaults = []
+ssl = ["tokio-openssl", "openssl"]
```
[the remark this answers is not in the document]
Comment (psarna) [reply scylla-cql/Cargo.toml:14]: A random google search found this: https://github.com/est31/cargo-udeps - maybe we should give it a try in the top-level crate too

[the remark this answers is not in the document]
Comment (wprzytula) [reply scylla-cql/Cargo.toml]: The crate has shown to be truly useful - it told me I have missed one unused dependency during manual removal.

File: scylla-cql/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+keywords = ["database", "scylla", "cql", "cassandra"]
+categories = ["database"]
+license = "MIT OR Apache-2.0"
+
+[dependencies]
+scylla-macros = { version = "0.1.1", path = "../scylla-macros"}
+byteorder = "1.3.4"
+bytes = "1.0.1"
+num_enum = "0.5"
+tokio = { version = "1.12", features = ["net", "time", "io-util", "sync", "rt", "macros"] }
```
Comment (piodul) [inline scylla-cql/Cargo.toml:17]: Please remove the features from the list. I checked and we only use `AsyncRead`, `AsyncReadExt` and they don't need any of the tokio features enabled.

Just

```toml
tokio = "1.12"
```

should suffice.

**Review by piodul — APPROVED** [review]

**Pull request merged**